src/Orleans/Async/Cancellation/ICancellationSourcesExtension.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    internal interface ICancellationSourcesExtension : IGrainExtension, IGrain
+    {
+        [AlwaysInterleave]
+        Task CancelTokenSource(GrainCancellationToken token);
```

Comment:
> So now there's a case when cancel request arrives before the original one and fails

This case is not possible.
If the user sends cancel before he sent the actual token, its a bug in his code.
If he send them in the right order but did not first await the first call - again, its a bug in the code, as all casually related operations should be expressed explicitly via await.